src/Resource/Collection.js

```diff
@@ -4,32 +4,42 @@ import ParameterBag from './ParameterBag';
 /**
  * @param {String|Object} name
  * @param {HyralRepository} repository
- * @param {Object} serializedData
+ * @param {Object} dataDefaults
+ * @param {ParameterBag|Object} parameterBag
  */
-function Collection(name, repository, serializedData = {
+function Collection(
   name,
-  items: [],
-  metadata: {
-    loaded: false,
-    parameterBagData: null,
-    paging: {
-      count: null,
-      pages: null,
+  repository,
+  dataDefaults = {
+    items: [],
+    metadata: {
+      loaded: false,
+      paging: {
+        count: null,
+        pages: null,
+      },
     },
   },
-}) {
+  parameterBag = null,
```

Comment:
Ik begrijp de injection van de ParameterBag. Ik zat ook al met het feit dat er nu in de Collection een ParameterBag wordt aangemaakt. 
Misschien moet het wel zo, maar ik twijfel er toch over. 
In ieder geval hebben we een functie nodig die de collection vrij eenvoudig deserialiseerd:

`createCollectionFromData(data) -> Collection`